### PR TITLE
Skills cooldowns

### PR DESCRIPTION
**Pull request by AminArria**
- Check skill cooldown when activating
- Add skill cooldown to `player.cooldowns` when activating
- Reduce cooldowns on game tick

### CODE REVIEW

File: native/lambda_game_engine/src/game.rs [hunk 1]
```diff
@@ -387,6 +397,15 @@ fn update_player_actions(players: &mut HashMap<u64, Player>, elapsed_time_ms: u6
...
 
+fn update_player_cooldowns(players: &mut HashMap<u64, Player>, elapsed_time_ms: u64) {
+    players.values_mut().for_each(|player| {
+        player
+            .cooldowns
+            .iter_mut()
+            .for_each(|(_key, remaining)| *remaining = remaining.saturating_sub(elapsed_time_ms))
+    })
+}
+
```
[the remark this answers is not in the document]
Comment (mFragaBA) [reply native/lambda_game_engine/src/game.rs:408]: something like this maybe?
```suggestion
fn update_player_cooldowns(players: &mut HashMap<u64, Player>, elapsed_time_ms: u64) {
    players.values_mut().for_each(|player| {
        player
            .cooldowns
            .retain(|_key, remaining| {
                *remaining = remaining.saturating_sub(elapsed_time_ms);
                *remaining > 0
            })
}

```

[the remark this answers is not in the document]
Comment (mFragaBA) [reply native/lambda_game_engine/src/game.rs:408]: I checked and with the current implementation I can only shoot once and then I can't anymore

**Review by mFragaBA — CHANGES_REQUESTED** [review]
Nice work! I left a small comment regarding clearing the cooldowns. I think with that this should be good to merge

File: native/lambda_game_engine/src/game.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AminArria) [reply native/lambda_game_engine/src/game.rs]: Done

**Review by manucamejo — APPROVED** [review]

**Review by mFragaBA — APPROVED** [review]
LGTM!

**Pull request merged**